data/index.js

```diff
@@ -241,6 +241,10 @@ export const withSelect = ( mapStateToProps ) => createHigherOrderComponent( ( W
 			this.state = {};
 		}
 
+		shouldComponentUpdate( nextProps, nextState ) {
```

Comment:
React redux is pure by default, This ensures `withSelect` is pure as well. I guess this is needed when the parent component rerenders but passing the exact same props for one of its children components.

Reply:
I'm wondering if we should do the same for the `withDispatch` HoC

Reply:
Rather than in both, I'm inclined to think that it should be in neither, and that we offer up a "pure" standalone higher-order component. Nothing about `withSelect` or `withDispatch` makes me think it should be providing this functionality for anything other than the own props it merges.

Reply:
Sure, it just seems so arbitrary to add it here. Why not any other of the higher-order components?

Reply:
It is arbitrary, the same way it's arbitrary in `connect`. Probably not a great argument, but my idea was to match `connect` (`withSelect` is the closest thing to `connect`). Also, this is probably the most used HoC